### PR TITLE
Fix/AJD-658 abnormal longitudinal speed

### PR DESCRIPTION
**Pull request by dmoszynski**
## Types of PR

- [ ] New Features
- [ ] Upgrade of existing features
- [x] Bugfix

## Description
Problem:
image

The problem was related to the sometimes missing initialization of the `target_seed` variable during the tick of `MoveBackwordAction`.

- I’ve added a check before using it, if the variable is already initialized.
- If it isn't then it is assigned a value with is calculated on the basis of previous lanelets - because it's `MoveBackwordAction`.
- The changes I’ve made are based on other actions implemented in the `behavior_tree_plugin` package (there are analogous checks).

Details and tests: https://tier4.atlassian.net/browse/AJD-658?focusedCommentId=112178

## Others
Regression tests: 
https://tier4.atlassian.net/browse/AJD-658?focusedCommentId=112334
https://tier4.atlassian.net/browse/AJD-658?focusedCommentId=112437
https://tier4.atlassian.net/browse/AJD-658?focusedCommentId=112801

### CODE REVIEW

File: simulation/behavior_tree_plugin/src/vehicle/follow_lane_sequence/move_backward_action.cpp [hunk 1]
```diff
@@ -75,6 +75,11 @@ BT::NodeStatus MoveBackwardAction::tick()
   if (waypoints.waypoints.empty()) {
     return BT::NodeStatus::FAILURE;
   }
+  auto following_lanelets =
```
Comment (hakuturu583) [inline simulation/behavior_tree_plugin/src/vehicle/follow_lane_sequence/move_backward_action.cpp:78]: Isn't it a previous_lanelet?

File: simulation/behavior_tree_plugin/src/vehicle/follow_lane_sequence/move_backward_action.cpp [hunk 2]
```diff
@@ -75,6 +75,11 @@ BT::NodeStatus MoveBackwardAction::tick()
...
+    hdmap_utils->getPreviousLanelets(entity_status.lanelet_pose.lanelet_id);
+  if (!target_speed) {
+    target_speed = hdmap_utils->getSpeedLimit(following_lanelets);
```
Comment (hakuturu583) [inline simulation/behavior_tree_plugin/src/vehicle/follow_lane_sequence/move_backward_action.cpp:81]:
```
target_speed = hdmap_utils->getSpeedLimit(
  hdmap_utils->getPreviousLanelets(entity_status.lanelet_pose.lanelet_id));
```

Is better.

**Review by hakuturu583 — CHANGES_REQUESTED** [review]

**Comment by hakuturu583** [comment]
@dmoszynski Please use

ament_clang_format --reformat
before pushing your code.

**Comment by HansRobo** [comment]
the command `ament_clang_format --reformat` doesn't fix errors related to newlines, like `Replace [\n    ] with [ ]`
```
 [src/vehicle/follow_lane_sequence/move_backward_action.cpp:78:27]: Replace [\n    ] with [ ]
    -   auto following_lanelets =
    -     hdmap_utils->getPreviousLanelets(entity_status.lanelet_pose.lanelet_id);
    +   auto following_lanelets = hdmap_utils->getPreviousLanelets(entity_status.lanelet_pose.lanelet_id);
```
I recommend that you manually fix such errors as the error instructed.
Anyway, green ci is needed for merging into master!

**Review by dmoszynski — COMMENTED** [review]
@hakuturu583 Corrected

**Comment by hakuturu583** [comment]
@dmoszynski Thanks, it was bug.

**Review by hakuturu583 — APPROVED** [review]
LGTM

**Pull request merged**